src/dashboard/infrastructure/analytics-4/site-kit-analytics-4-adapter.php

```diff
@@ -0,0 +1,240 @@
+				// As per https://developers.google.com/analytics/devguides/reporting/data/v1/basics#read_the_response,
+				// the order of the columns is consistent in the request, header, and rows.
+				// So we can use the key of the header to get the correct metric value from the row.
+				$metric_value = $daily_traffic->metricValues[ $key ]->value;
+
+				if ( $metric->name === 'sessions' ) {
+					$traffic_data->set_sessions( (int) $metric_value );
+				}
+				elseif ( $metric->name === 'totalUsers' ) {
+					$traffic_data->set_total_users( (int) $metric_value );
+				}
+			}
+
+			// Since we're here, we know that the first dimension is date, so we know that dimensionValues[0]->value is a date.
+			$data_container->add_data( new Daily_Traffic_Data( $daily_traffic->dimensionValues[0]->value, $traffic_data ) );
+		}
+
+		return $data_container;
+	}
+
+	/**
+	 * Parses a response for a Site Kit API request for Analytics 4 that compares data ranges.
+	 *
+	 * @param mixed $response The response to parse.
+	 *
+	 * @return Data_Container The parsed response.
+	 */
+	protected function parse_comparison_response( $response ): Data_Container {
+		$data_container          = new Data_Container();
+		$comparison_traffic_data = new Comparison_Traffic_Data();
+
+		// First row is the current date range's data, second row is the previous date range's data.
+		foreach ( $response->rows as $date_range_key => $date_range_row ) {
+			$traffic_data = new Traffic_Data();
+
+			// Loop through all the metrics of the date range.
+			foreach ( $response->metricHeaders as $key => $metric ) {
+
+				// @TODO: Maybe use class methods like MetricValue::getValue() instead of ->value.
+				// As per https://developers.google.com/analytics/devguides/reporting/data/v1/basics#read_the_response,
+				// the order of the columns is consistent in the request, header, and rows.
+				// So we can use the key of the header to get the correct metric value from the row.
+				$metric_value = $date_range_row->metricValues[ $key ]->value;
+
+				if ( $metric->name === 'sessions' ) {
+					$traffic_data->set_sessions( (int) $metric_value );
+				}
+				elseif ( $metric->name === 'totalUsers' ) {
+					$traffic_data->set_total_users( (int) $metric_value );
+				}
+			}
+
+			if ( $date_range_key === 0 ) {
+				$comparison_traffic_data->set_current_traffic_data( $traffic_data );
+			}
+			elseif ( $date_range_key === 1 ) {
+				$comparison_traffic_data->set_previous_traffic_data( $traffic_data );
+			}
+		}
+
+		$data_container->add_data( $comparison_traffic_data );
+
+		return $data_container;
+	}
+
+	/**
+	 * Checks the response of the request to detect if it's a comparison request.
+	 *
+	 * @param mixed $response The response.
+	 *
+	 * @return bool Whether it's a comparison request.
+	 */
+	protected function is_comparison_request( $response ): bool {
+		if ( \count( $response->dimensionHeaders ) === 1 && $response->dimensionHeaders[0]->name === 'dateRange' ) {
```

Comment:
Why not just return (count && )?

Reply:
LOL, right, silly me 😅 [Done](https://github.com/Yoast/wordpress-seo/pull/22023/commits/d4289e5b07c9e0e086cb9a0c85032d763d6aaa4b)